src/include/simeng/OS/Constants.hh

```diff
@@ -122,8 +225,42 @@ static constexpr bool hasFault(uint64_t value) {
 static constexpr uint64_t getFaultCode(uint64_t value) {
   if ((value & masks::faults::pagetable::FAULT) != pagetable::FAULT)
     return masks::faults::pagetable::NO_FAULT;
-  return (0x7000000000000000 & value) & 0xF000000000000000;
-};
+  return (0x7000000000000000 & value);
+}
+
+/** Function to print the fault code in a human friendly manner. */
```

Comment:
You've done this by passing the prefix and then outputting. May be better the other way round, passing the human-readable fault error msg back to the caller so we can have more control of what happens with it (e.g. output to cout vs cerr)